src/Veldrid/GraphicsApiVersion.cs

```diff
@@ -0,0 +1,67 @@
+﻿
+namespace Veldrid
+{
+    public readonly struct GraphicsApiVersion
+    {
+        public static GraphicsApiVersion Unknown => default;
+
+        public int Major { get; }
+        public int Minor { get; }
+        public int Subminor { get; }
+        public int Patch { get; }
+
+        public bool IsKnown => Major != 0 && Minor != 0 && Subminor != 0 && Patch != 0;
+
+        public GraphicsApiVersion(int major, int minor, int subminor, int patch)
+        {
+            Major = major;
+            Minor = minor;
+            Subminor = subminor;
+            Patch = patch;
+        }
+
+        public override string ToString()
+        {
+            return $"{Major}.{Minor}.{Subminor}.{Patch}";
+        }
+
+        /// <summary>
+        /// Parses OpenGL version strings with either of following formats:
+        /// <list type="bullet">
+        ///   <item>
+        ///     <description>major_number.minor_number</description>
+        ///   </item>
+        ///   <item>
+        ///     <description>major_number.minor_number.release_number</description>
+        ///   </item>
+        /// </list>
+        /// </summary>
+        /// <param name="versionString">The OpenGL version string.</param>
+        /// <param name="version">The parsed <see cref="GraphicsApiVersion"/>.</param>
+        /// <returns>True whether the parse succeeded; otherwise false.</returns>
+        public static bool TryParseGLVersion(string versionString, out GraphicsApiVersion version)
```

Comment:
I don't think this needs to be a public method.

Reply:
Hmm I had a reason for making this public but I forgot it. Anyways, this parses based on OpenGL specification so it shouldn't ever change behavior-wise.